### DISCUSSION TITLE
Real-Time Notification Updates

### DISCUSSION THREAD

**Discussion by paesjr** [directus/directus#21302] [Feature Requests]
### Summary

Implement Real-Time Notification Updates in the Directus App

### Motivation

The current lack of real-time features in the app, specifically regarding the Notification Bell, hinders a seamless user experience. Manual browser refreshes are required to view notifications, causing delays and potential user frustration. The motivation behind this feature request is to enhance user engagement and efficiency by introducing real-time capabilities, ensuring that users are promptly notified of relevant activities.

### Detailed Design

To implement real-time notification updates, the following steps should be considered:

Real-Time Notification System:
Develop a real-time notification system to enable instant updates.
Browser Compatibility:
Ensure compatibility with major browsers (e.g., Chrome, Firefox, Safari).
User Preferences:
Allow users to customize the frequency of real-time updates or provide an option to toggle between real-time and manual refresh based on user preferences.

### Must Have:
- Real-Time Notification System
- Compatibility with Major Browsers
### Should Have:
- User Preferences for Update Frequency
### Could Have:
- Integration with User Dashboard
### Won't Have:
- Integration with External Notification Services

### Drawbacks

While the introduction of real-time features brings significant benefits, there are potential drawbacks to consider:

Implementation cost in terms of code size and complexity.
Impact on teaching people Directus, necessitating user education on the new real-time features.
Potential migration costs for existing Directus applications, especially if this is considered a breaking change.

### Alternatives

Alternative designs considered include:

Polling mechanism for periodic updates.
Integration with external notification services.

### Adoption Strategy

To facilitate a smooth adoption process:

Provide clear documentation on the implementation of real-time features.
Consider backward compatibility for existing applications.
Offer a migration guide for users transitioning to the real-time notification system.

**Comment by erictong514** [comment]
This feature is very important to me.

How can the push notification function be displayed only after manual refresh? In this way, there is no push notification feature

**Comment by JamesW1** [comment]
Thanks for the suggestion! We've moved feature requests over to our public roadmap at https://roadmap.directus.io/roadmap

I've migrated this request there for you: https://roadmap.directus.io/roadmap?id=5f91ac1e-3bad-41d8-bf0f-f51e48ee5b77

Moving forwards, any updates on this subject will be visible on our public roadmap.
